web/src/api.ts

```diff
@@ -48,6 +49,29 @@ export const getListing = async (listingId: number): Promise<Listing> => {
   return res.json();
 };
 
+/**
+ * Retrieves merchant with the given email from the database.
+ */
+export const getMerchantWithEmail = async (
+  email: string
+): Promise<MerchantResponse> => {
+  const res = await fetch(
+    `${process.env.REACT_APP_SERVER_URL}/merchants?email=${email}`
+  );
+
+  if (res.status !== 200) {
+    throw new Error(SERVER_ERROR);
+  }
+
+  const merchants = await res.json();
+
+  if (merchants.length < 1) {
```

Comment:
When you add your merchant, did you have a check on the uniqueness of merchant email ? So confirm merchants.length<=1.

Reply:
Yes. I check for uniqueness of the merchant `firebaseUid` but the email is unique for different `firebaseUid`.